### PR TITLE
chore(docs): update v10 migration docs 

### PR DESCRIPTION
**Pull request by jnm2377**

### CODE REVIEW

**Review by lovemecomputer — APPROVED** [review]
cool, looks great!! i have one word-choice comment to make which i will make inline…

File: src/components/data-table/migrate-to-10.x.md [hunk 1]
```diff
@@ -1,20 +1,79 @@
 ## Data Table
 
+### HTML
+
+Aside from using new icons from the [`carbon-elements`](https://github.com/IBM/carbon-elements) package, there are several structural changes and new selectors in the Data Table component. Please see the new structure below and reference the [data table page](https://next.carbondesignsystem.com/components/data-table/code) in our site to copy the specified new markup.
+
+React and other framework variants should reflect the changes automatically.
```
Comment (vpicone) [inline src/components/data-table/migrate-to-10.x.md:7]: There's some changes that folks are going to have to make (toolbar/search related)

Comment (jnm2377) [reply src/components/data-table/migrate-to-10.x.md:7]: @vpicone What should I add specifically regarding that?

**Comment by tw15egan** [comment]
@vpicone is there anything @jnm2377 can add now, or should we create a separate issue for additional documentation regarding toolbar/search

**Comment by vpicone** [comment]
@tw15egan I think it's good now, we'd probably want to link to the React migration guide that doesn't exist yet.

**Review by vpicone — APPROVED** [review]

**Pull request merged**